apps/yggdrasil_web/web/controllers/character_controller.ex

```diff
@@ -2,13 +2,26 @@ defmodule YggdrasilWeb.CharacterController do
   use YggdrasilWeb.Web, :controller
   use Guardian.Phoenix.Controller
 
+  import YggdrasilWeb.EnsurePermission, only: [check_permissions: 2]
+
   alias Yggdrasil.Character
+  alias YggdrasilWeb.EnsurePermission
+
+  plug EnsurePermission, [character: [:read]] when action in [:index, :show]
```

Comment:
usage of new plug to enforce permissions.

Still need to wrap some tests around it but all tests currently pass that we have.